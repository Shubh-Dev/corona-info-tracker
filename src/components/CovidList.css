.main-container {
  display: grid;
  grid-template-columns: 1fr 1fr;
  color: #fff;
}

.main-headline {
  font-size: 25px;
  font-family:
    Roboto,
    -apple-system,
    BlinkMacSystemFont,
    'Segoe UI',
    'Oxygen',
    'Ubuntu',
    'Cantarell',
    'Fira Sans',
    'Droid Sans',
    'Helvetica Neue',
    sans-serif;
  margin-left: 1rem;
}

.header {
  display: flex;
  justify-content: space-between;
}

.searchbar {
  width: 10rem;
  height: 25px;
  border: none;
  border-radius: 5px;
  margin: auto 10px auto 10px;
}

.container {
  padding-right: 10px;
  margin: auto;
}
