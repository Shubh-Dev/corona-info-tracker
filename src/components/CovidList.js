import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
// import { generate } from 'randomized-string';
import { fetchCovidData } from '../redux/covidslice/covidSlice';
import Covid from './Covid';
import './CovidList.css';

const CovidList = () => {
  const [searchTerm, setSearchTerm] = useState('');

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCovidData());
  }, []);

  const { items } = useSelector((state) => state.items);

  /* eslint-disable */
  return (
    <div className="container">
      <header className="header">
        <h1 className="main-headline">Covid 19 Statistics</h1>
        <input
          className="searchbar"
          type="searchbar"
          placeholder="Search by country name.."
          onChange={(event) => { setSearchTerm(event.target.value); }}
        />
      </header>
      <div className="main-container">
        {items.filter((val) => {
          if (searchTerm === '') {
            return val;
          } if (val.country.toLowerCase().includes(searchTerm.toLowerCase())) {
            return val;
          }
        }).map(({
           country, flag, cases,
        }) => (
          <Covid
            // key={generate()}
            country={country}
            flag={flag}
            cases={cases}
          />
        ))}
      </div>
    </div>
  );
};

export default CovidList;
