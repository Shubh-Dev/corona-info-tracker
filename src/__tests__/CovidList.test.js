import React from 'react';
import CovidList from '../components/CovidList';
import '@testing-library/jest-dom'
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore'

it('CovidList render correctly', () => {
  const Tree = TestRenderer.create(
    <Provider store={store}>
        <CovidList /> 
    </Provider>
  );
  expect(Tree).toMatchSnapshot();
})
