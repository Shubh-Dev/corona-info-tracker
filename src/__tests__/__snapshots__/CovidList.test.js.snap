// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CovidList render correctly 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <h1
      className="main-headline"
    >
      Covid 19 Statistics
    </h1>
    <input
      className="searchbar"
      onChange={[Function]}
      placeholder="Search by country name.."
      type="searchbar"
    />
  </header>
  <div
    className="main-container"
  />
</div>
`;
